python/pyspark/sql/types.py

```diff
@@ -186,6 +186,30 @@ def fromInternal(self, ts):
             return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
 
 
+class CalendarIntervalType(DataType, metaclass=DataTypeSingleton):
+    """Calendar Interval type
+    """
+    @classmethod
+    def typeName(cls):
+        return "interval"
+
+    def needConversion(self):
+        return True
+
+    def simpleString(self):
+        return "interval"
+
+    def toInternal(self, di):
+        raise NotImplementedError(
+            "Conversion from external Python types to interval not supported"
+        )
```

Comment:
I suppose in the future if we want to support conversion of Python's [`datetime.timedelta`](https://docs.python.org/3/library/datetime.html#datetime.timedelta), it would happen here, right?

Reply:
For full support we might need both Python and JVM component. If I recall correctly `timedelta` has razrovine mapping to their internal `net.razorvine.pickle.objects.TimeDelta`.

In the opposite direction we could, if I am not mistaken, start with making `CalendarInterval` bean compatible, but there is compatibility issue ‒ we'd have to map from Spark's months to Python's days.